documentation/src/docs/metrics.md

```diff
@@ -7,491 +7,521 @@ You may obtain a copy of the License at
-segmentstore.segment.read_bytes
-segmentstore.segment.write_events
-```
+2. **Component Origin**: Indicates which component generates the metric, such as `segmentstore`, `controller`.
 
-- Segment Store cache read/write latency metrics (histogram):
-```
-segmentstore.cache.insert_latency_ms
-segmentstore.cache.get_latency
-```
+3. **Sub-Component (or Abstraction)**: Indicates the second level component or abstraction, such as `cache`, `transaction`, `storage`.
 
-- Segment Store cache read/write metrics (counters):
-```
-segmentstore.cache.write_bytes
-segmentstore.cache.read_bytes
-```
+4. **Metric Base Name**: Indicates the `read_latency_ms`, `create_count`.
 
-- Segment Store cache size (gauge) and generation spread (histogram) metrics:
-```
-segmentstore.cache.size_bytes
-segmentstore.cache.gen
-```
+For example:
+ ```
+     pravega.segmentstore.segment.create_latency_ms
+  ```
+Following are some common combinations of component and sub-components (or abstractions) being used:
 
-- Tier-1 DurableDataLog read/write latency and queueing metrics (histogram):	
-```
-segmentstore.bookkeeper.total_write_latency_ms
-segmentstore.bookkeeper.write_latency_ms
-segmentstore.bookkeeper.write_queue_size
-segmentstore.bookkeeper.write_queue_fill
-```
+- `segmentstore.segment`: Metrics for individual Segments
+- `segmentstore.storage`: Metrics related to long-term storage (Tier 2)
+- `segmentstore.bookkeeper`: Metrics related to Bookkeeper (Tier 1)
+- `segmentstore.container`: Metrics for Segment Containers
+- `segmentstore.thread_pool`: Metrics for Segment Store thread pool
+- `segmentstore.cache`: Cache-related metrics (RocksDB)
+- `controller.stream`: Metrics for operations on Streams (e.g., number of streams created)
+- `controller.segments`: Metrics about Segments, per Stream (e.g., count, splits, merges)
+- `controller.transactions`: Metrics related to Transactions (e.g., created, committed, aborted)
+- `controller.retention`: Metrics related to data retention, per Stream (e.g., frequency, size of truncated data)
+- `controller.hosts`: Metrics related to Pravega servers in the cluster (e.g., number of servers, failures)
+- `controller.container`: Metrics related to Container lifecycle (e.g., failovers)
 
-- Tier-1 DurableDataLog read/write (counter) and per-container ledger count metrics (gauge):	
-```
-segmentstore.bookkeeper.write_bytes
-segmentstore.bookkeeper.bookkeeper_ledger_count - with tag {"container", $containerId}
-```
+Following are the two types of metrics:
 
-- Tier-2 Storage read/write latency metrics (histogram):	
-```
-segmentstore.storage.read_latency_ms
-segmentstore.storage.write_latency_ms
-```
+1. **Global Metric**: Values are directly associated to the metric name that appears in this file. These can be used when we want to report metric values that applies to the whole Pravega cluster (e.g., number of bytes written, operations).
+For instance, `STORAGE_READ_BYTES` can be classified as a Global metric.
```

Comment:
@jkhalack In this case, if we sum up the metrics of all segmentstore (or ignore the host tag) then we could have a global view, and it is indeed the purpose of the metrics. The doc is just trying to differentiate two kinds of metrics, those per stream/segment/container, and those "global".

Reply:
Still, I feel that it is wrong to say that the `global` metrics describe pravega cluster as a whole.
They do describe a single pravega component only.

Reply:
I have made a lot of mistakes trying to interpret pravega metrics (just because I did not realize that there were three independent sets of metrics coming from three different segmentstore instances, and I needed to aggregate over all of them rather than treat each measurement as a single data value).

Therefore I could see other people repeating the mistakes as I did. And I would not want anyone else to spend more time than needed on trying to figure out what is in fact reported here.

Reply:
@Aparnarr @jkhalack - I suggest the following changes:
We have two types of metrics:
- Global metric: those tend to give a "global" view of some concern, by reporting values with host tag only (to retrieve the aggregated/global value, just drop the host tag in queries) ... ...

Reply:
I do not agree that simply dropping the host tag would be enough.
If the query is looking for `mean` aggregate, one would end up with getting the mean value over segmentstore  or over controller instances, instead of getting a cluster global averaged over time.

My suggestion is to mention that `_global` metrics are reporting global values  per component (_segmentstore_  or _controller_) instance, and further aggregation logic is needed if looking for pravega cluster globals.

Reply:
Well said, I agree.

Reply:
Accepted changes